### PR TITLE
Update configuration and dependencies

### PR DESCRIPTION
**Pull request by ganeshdipdumbare**
- update config to use viper

## Summary by CodeRabbit

- **New Features**
	- Enhanced configuration management by integrating the `viper` library, simplifying the loading of configuration parameters such as database URL and server port.
	- Introduced validation for configuration settings to ensure proper values are set for parameters like delays and retries.

- **Bug Fixes**
	- Streamlined the configuration loading process, reducing potential errors from custom environment variable handling.

- **Chores**
	- Updated and added several dependencies to improve project maintainability and functionality.
	- Added new tests for configuration validation to ensure robustness against various edge cases.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -22,36 +22,42 @@ type Config struct {
 }
 
 func LoadConfig() (*Config, error) {
+	v := viper.New()
+
+	// Set defaults
+	v.SetDefault("database_url", "postgresql://user:password@localhost:5432/qonto_accounts?sslmode=disable")
```
[the remark this answers is not in the document]
Comment (ganeshdipdumbare) [reply config/config.go:47]: please ignore this one as it is test repo
